collection.go

```diff
@@ -61,8 +61,8 @@ func (cs *CollectionSpec) Copy() *CollectionSpec {
 	}
 
 	cpy := CollectionSpec{
-		Maps:      make(map[string]*MapSpec, len(cs.Maps)),
-		Programs:  make(map[string]*ProgramSpec, len(cs.Programs)),
+		Maps:      copyMapOfSpecs(cs.Maps),
```

Comment:
Did you forget to remove the two `for name, spec := range cs... {` loops below? I think this copies twice now.

Reply:
Derp :cry: